File: angular-crud/src/app/service/employee.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Observable, catchError } from 'rxjs';
import { Employee } from '../model/employee.model';

@Injectable({
  providedIn: 'root'
})
export class EmployeeService {

  private baseUrl = 'http://localhost:8080/api/employee';

  constructor(private http: HttpClient) {}

  httpOptions = {
    headers: new HttpHeaders({
      'Content-Type': 'application/json'
    })
  }

  getAllEmployees(): Observable<any> {
    return this.http.get(this.baseUrl).pipe(
      catchError(this.handleError)
    );
  }

  getEmployeeById(id: string): Observable<any> {
    return this.http.get(`${this.baseUrl}/${id}`).pipe(
      catchError(this.handleError)
    );
  }

  createEmployee(employee: Employee): Observable<any> {
    return this.http.post(this.baseUrl, employee).pipe(
      catchError(this.handleError)
    );
  }

  updateEmployee(id: string, employee: Employee): Observable<any> {
    return this.http.put(`${this.baseUrl}/${id}`, employee).pipe(
      catchError(this.handleError)
    );
  }

  deleteEmployee(id: string): Observable<any> {
    return this.http.delete(`${this.baseUrl}/${id}`).pipe(
      catchError(this.handleError)
    );
  }


  private handleError(error: any): Promise<any> {
    console.error('An error occurred', error);
    return Promise.reject(error.message || error);
  }

}

File: springboot-crud/src/main/java/com/misoft/crud/service/EmployeeService.java
package com.misoft.crud.service;

import com.misoft.crud.DTO.EmployeeDTO;
import com.misoft.crud.model.Employee;
import org.springframework.http.ResponseEntity;

import java.util.List;

public interface EmployeeService {

    public ResponseEntity<List<Employee>> getAllEmployees();
    public ResponseEntity<EmployeeDTO> getEmployeeById(String id);
    public ResponseEntity<String> createEmployee(EmployeeDTO employeeDTO);
    public ResponseEntity<String> updateEmployee(String id, EmployeeDTO employeeDTO);
    public ResponseEntity<String> deleteEmployee(String id);

}
